main.tf

```diff
@@ -70,13 +70,30 @@ module "efs" {
   vpc_id             = "${var.vpc_id}"
   subnets            = "${var.private_subnets}"
   availability_zones = "${var.availability_zones}"
-  security_groups    = ["${module.elastic_beanstalk_environment.security_group_id}"]             # EB/EC2 instances are allowed to connect to the EFS
   zone_id            = "${var.zone_id}"
+
+  # EB/EC2 instances and DataPipeline instances are allowed to connect to the EFS
+  security_groups = ["${module.elastic_beanstalk_environment.security_group_id}", "${module.efs_backup.security_group_id}"]
```

Comment:
By-default Amazon allows 1 SG per network interface.
Won't it cause errors?
Would it be better, if we allow [terraform-aws-efs-backup](https://github.com/cloudposse/terraform-aws-efs-backup) to add `Security Group Rule` instead of adding a `Security Group`?

Reply:
It has nothing to do with Network Interfaces, we are dealing with just SGs (firewalls) here.
A Security Group can have many inbound rules, and each rule can accept another security group.

In `Terraform`, this is expressed as 

```
resource "aws_security_group" "default" {
  name        = "${module.label.id}"
  description = "EFS"
  vpc_id      = "${var.vpc_id}"

  ingress {
    from_port       = "2049"                     # NFS
    to_port         = "2049"
    protocol        = "tcp"
    security_groups = ["${var.security_groups}"]
  }
```
https://www.terraform.io/docs/providers/aws/r/security_group.html#security_groups

So we allowing the two external SGs (`EB` and `DataPipeline`) to connect to the `EFS` on port `2049`.

Which I suppose will do the same thing as `allow terraform-aws-efs-backup to add Security Group Rule`.

 `terraform-aws-efs-backup` `README.md` needs to be improved because, for example, it says:
> Integration with EFS
It's necessary to configure your EFS filesystem security groups to permit backups from the DataPipeline instances.
Add the security group ID from the efs_backup_security_group output to a security group of EFS Filesystems.

Which implies it's a requirement, but it's not.
The same effect could be achieved by setting `modify_security_group` to `true`, but the description for the attribute is confusing:

> Should the module modify EFS security groups (if set to false backups will fail)

And at the same time, the default value is set to `false`.

Ideally, `README.md` should say:
There are two ways of enabling connectivity between the `EFS` and the `DataPipeline` instances. Use one of the following:
1. Explicitly add the `DataPipeline` SG to the list of the `ingress` rules of the `EFS` SG
2. Set `modify_security_group` to `true` so the module will modify the `EFS` SG by adding a rule to allow the `DataPipeline` SG to connect to the `EFS` SG

Reply:
seems reasonable to me to make this change. @comeanother thoughts?

Reply:
Please add PR to [terraform-aws-efs-backup](https://github.com/cloudposse/terraform-aws-efs-backup) with suggested description update.